### PR TITLE
style-guide: update page

### PR DESCRIPTION
**Pull request by kbdharun**
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).

## Changes

- Add a special case for enclosing mnemonic terms inside brackets (Ref. #12151)
- Revise the entire option syntax section (implementing suggestions from #12125)
- Minor fixes

### CODE REVIEW

File: contributing-guides/style-guide.md [hunk 1]
```diff
@@ -287,13 +287,15 @@ Use backticks on the following:
 - Commands, e.g. `ls`.
 - Standard streams: `stdout`, `stdin`, `stderr`. **Do not** use the full names (e.g. standard output).
```
Comment (vitorhcl) [inline contributing-guides/style-guide.md:288]:
```suggestion
- Standard streams: `stdout`, `stdin`, `stderr`. **Do not** use the full names (e.g. standard output).
- Compression algorithms, e.g. `zip`, `7z`, `xz`.
```
This is a common case for backticks

File: contributing-guides/style-guide.md [hunk 2]
```diff
@@ -287,13 +287,15 @@ Use backticks on the following:
...
 
+- Optionally, Mnemonics and their enclosed terms can be separated with brackets from the rest of the description (i.e. `([a]ll)`) in translations and specific pages to provide additional context or mention a word not present in the description.
```
Comment (vitorhcl) [inline contributing-guides/style-guide.md:290]:
```suggestion
- Optionally, mnemonics and their enclosed terms can be separated with brackets from the rest of the description (i.e. `([a]ll)`) in translations and specific pages to provide additional context or mention a word not present in the description.
```
I think it would be easier to find this if it was on the mnemonics section

**Review by vitorhcl — APPROVED** [review]
Thanks for your contribution! I have some suggestions

**Review by vitorhcl — APPROVED** [review]

**Comment by kbdharun** [comment]
IG I will try to address the https://github.com/tldr-pages/tldr/issues/12125 (long option) suggestion in this PR too, will make a commit to update it.

**Comment by vitorhcl** [comment]
> IG I will try to address the https://github.com/tldr-pages/tldr/issues/12125 (long option) suggestion in this PR too, will make a commit to update it.

For consistency, I would suggest that it should be emphatisized that this should only be used when short options are preferred over long options on the rest of the page.

**Comment by kbdharun** [comment]
> > IG I will try to address the #12125 (long option) suggestion in this PR too, will make a commit to update it.
> 
> For consistency, I would suggest that it should be emphatisized that this should only be used when short options are preferred over long options on the rest of the page.

This is the rough draft I worked on for the option syntax section:

```
## Option syntax

- For commonly/frequently used commands (e.g. `grep`, `tar`, `etc`), we prefer using short options along with[mnemonics](#short-option-mnemonics).
- For highlighting, both long and short options combine them within a placeholder i.e. `-{{o|-output}}`.
- Use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms) for pages in `common` directory.
- When documenting PowerShell commands, use **PowerShell-style long options** (like `-Help` instead of `-H`).
- We prefer using a space instead of the equals sign (`=`) to separate options from their arguments (i.e. use `--opt arg` instead of `--opt=arg`) unless the program does not support it.
```

What do you think about this?

File: contributing-guides/style-guide.md [hunk 3]
```diff
@@ -182,15 +182,12 @@ In this case, provide a note and method to determine whether the command current
 
 ## Option syntax
 
-- Use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms).
+- For commonly/frequently used commands (e.g. `grep`, `tar`, `etc`), we prefer using short options along with[mnemonics](#short-option-mnemonics).
+- For highlighting, both long and short options combine them within a placeholder i.e. `-{{o|-output}}`.
```
Comment (vitorhcl) [inline contributing-guides/style-guide.md:186]:
```suggestion
- For highlighting both long and short options, combine them within a placeholder i.e. `-{{o|-output}}`.
```

File: contributing-guides/style-guide.md [hunk 4]
```diff
@@ -182,15 +182,12 @@ In this case, provide a note and method to determine whether the command current
 
 ## Option syntax
 
-- Use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms).
+- For commonly/frequently used commands (e.g. `grep`, `tar`, `etc`), we prefer using short options along with[mnemonics](#short-option-mnemonics).
```
Comment (vitorhcl) [inline contributing-guides/style-guide.md:185]:
```suggestion
- For commonly/frequently used commands (e.g. `grep`, `tar`, `etc`), we prefer using short options along with [mnemonics](#short-option-mnemonics).
```

File: contributing-guides/style-guide.md [hunk 5]
```diff
@@ -182,15 +182,12 @@ In this case, provide a note and method to determine whether the command current
...
+- For highlighting, both long and short options combine them within a placeholder i.e. `-{{o|-output}}`.
+- Use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms) for pages in `common` directory.
```
Comment (vitorhcl) [inline contributing-guides/style-guide.md:187]:
```suggestion
- For user-friendliness, use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms) for pages in `common` directory.
```
I think it's good to leave this documented.

**Review by vitorhcl — APPROVED** [review]
LGTM, I have some minor suggestions for improvements and typos

File: contributing-guides/style-guide.md [hunk 6]
```diff
@@ -182,15 +182,12 @@ In this case, provide a note and method to determine whether the command current
 
 ## Option syntax
 
-- Use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms).
+- For commonly/frequently used commands (e.g. `grep`, `tar`, `etc`), we prefer using short options along with [mnemonics](#short-option-mnemonics).
+- For highlighting both long and short options, combine them within a placeholder i.e. `-{{o|-output}}`.
```
Comment (acuteenvy) [inline contributing-guides/style-guide.md:186]: If we put the first hyphen outside, that kind of implies the long option is `---output` (3 hyphens).
```suggestion
- For highlighting both long and short options, combine them within a placeholder i.e. `{{-o|-output}}`.
```

**Comment by acuteenvy** [comment]
> For commonly/frequently used commands (e.g. `grep`, `tar`, `etc`), we prefer using short options

What are the disadvantages of adding `{{-short|--long-option}}` to every page?

File: contributing-guides/style-guide.md (hunk 6 above)
Comment (vitorhcl) [reply contributing-guides/style-guide.md:186]: Actually no, because there is one inside and one outside, which makes it 2 hypens

Comment (acuteenvy) [reply contributing-guides/style-guide.md:186]: Oh, I don't know what I was thinking.

How about this then?
```suggestion
- For highlighting both long and short options, combine them within a placeholder i.e. `{{-o|--output}}`.
```

One hyphen outside looks kind of ambiguous in a client at first glance.
screenshot

File: contributing-guides/style-guide.md [hunk 7]
```diff
@@ -182,15 +182,12 @@ In this case, provide a note and method to determine whether the command current
...
+- For user-friendliness, use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms) for pages in `common` directory.
 - When documenting PowerShell commands, use **PowerShell-style long options** (like `-Help` instead of `-H`).
-- When long options aren't available for a command, use **short options** instead.
-- While we prefer long options, we allow special cases in commands like `pacman` where short options are widely used and preferred over the long options (for cases like these decisions will be made by the maintainers on a case-by-case basis).
 - We prefer using a space instead of the equals sign (`=`) to separate options from their arguments (i.e. use `--opt arg` instead of `--opt=arg`) unless the program does not support it.
 
-> [!NOTE]
```
Comment (CleanMachine1) [inline contributing-guides/style-guide.md:191]: why is this being deleted.

Comment (kbdharun) [reply contributing-guides/style-guide.md:191]: I initially added this to highlight using long options and short option mnemonic. This point is being replaced by the first and second which explains this much concisely.

**Comment by kbdharun** [comment]
> What are the disadvantages of adding {{-short|--long-option}} to every page?

Let's take the example of a few technical pages like Gcloud and AWS, commands are often very long adding this too would make it exceed our character limit in a lot of cases. Also, updating all the pages to the new structure would be quite the undertaking but we could do it for commonly used commands i.e. package managers, etc.

**Comment by acuteenvy** [comment]
> Let's take the example of a few technical pages like Gcloud and AWS, commands are often very long adding this too would make it exceed our character limit in a lot of cases.

I agree with that, but it wouldn't hurt for short commands (most other pages).

File: contributing-guides/style-guide.md [hunk 8]
```diff
@@ -182,15 +182,12 @@ In this case, provide a note and method to determine whether the command current
 
 ## Option syntax
 
-- Use **GNU-style long options** (like `--help` rather than `-h`) when they are cross-platform compatible (intended to work the same across multiple platforms).
+- For commonly/frequently used commands (e.g. `grep`, `tar`, `etc`), we prefer using short options along with [mnemonics](#short-option-mnemonics).
```
Comment (acuteenvy) [inline contributing-guides/style-guide.md:185]: This sounds like the `{{-short|--long}}` syntax is not allowed on frequently used pages. Do we want to disallow it?

Comment (kbdharun) [reply contributing-guides/style-guide.md:185]: Nope, I will reword this to use wording instead.

File: contributing-guides/style-guide.md [hunk 9]
```diff
@@ -286,14 +285,15 @@ Use backticks on the following:
...
 - Brand and project names can be capitalized in the description whenever applicable (e.g. use `A tool for interacting with a Git repository.` instead of ``A tool for interacting with a `git` repository.``).
 - Acronym expansions (i.e. protocols, tools, etc) must not be translated unless there is a recognized native equivalent for them.
-- When documenting keycaps or a keyboard shortcut for a utility it is suggested to wrap them in backticks to make them stand out in the description (i.e. ``Print the last lines of a given file and keep reading file until `Ctrl + C`:``). Alternatively, you can document them as a separate command and optionally highlight them as placeholders (i.e. `:wq{{Enter}}` or `:wq<Enter>` or `:wq(Enter)`).
+- When documenting keycaps or a keyboard shortcut for a utility it is suggested to wrap them in backticks to make them stand out in the description (i.e. ``Print the last lines of a given file and keep reading the file until `Ctrl + C`:``). Alternatively, you can document them as a separate command and optionally highlight them as placeholders (i.e. `:wq{{Enter}}` or `:wq<Enter>` or `:wq(Enter)`).
```
Comment (vitorhcl) [inline contributing-guides/style-guide.md:296]:
```suggestion
- When documenting keycaps or a keyboard shortcut for a utility it is suggested to wrap them in backticks to make them stand out in the description (i.e. ``Print the last lines of a given file and keep reading it until `Ctrl + C`:``). Alternatively, you can document them as a separate command and optionally highlight them as placeholders (i.e. `:wq{{Enter}}` or `:wq<Enter>` or `:wq(Enter)`).
```

File: contributing-guides/style-guide.md [hunk 10]
```diff
@@ -400,8 +401,9 @@ Additionally, if the link is related to PowerShell command documentation, remove
 
 ## Help and version commands
 
-- We generally place the help and version commands at the **last part of the page** to highlight more practical commands at the beginning of the page.
+- We generally place the help and version commands at the **last part of the page** to highlight more practical commands at the beginning of the page. They can be replaced to accommodate other useful examples if required.
```
Comment (vitorhcl) [inline contributing-guides/style-guide.md:404]:
```suggestion
- We generally put, **in this order**, the help and version commands as the **last two** examples of the page to highlight more practical commands at the beginning of the page. They can be replaced to accommodate other useful examples if required.
```

**Review by vitorhcl — APPROVED** [review]
